logstash-core/src/main/java/org/logstash/common/io/FileCheckpointIO.java

```diff
@@ -0,0 +1,108 @@
+package org.logstash.common.io;
+
+import org.logstash.ackedqueue.Checkpoint;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class FileCheckpointIO  implements CheckpointIO {
+//    Checkpoint file structure
+//
+//    byte version;
+//    int pageNum;
+//    int firstUnackedPageNum;
+//    long firstUnackedSeqNum;
+//    long minSeqNum;
+//    int elementCount;
+
+    public static final int BUFFER_SIZE = Short.BYTES // version
+            + Integer.BYTES  // pageNum
+            + Integer.BYTES  // firstUnackedPageNum
+            + Long.BYTES     // firstUnackedSeqNum
+            + Long.BYTES     // minSeqNum
+            + Integer.BYTES  // eventCount
+            + Integer.BYTES;    // checksum
+
+    private final String dirPath;
+    private final String HEAD_CHECKPOINT = "checkpoint.head";
+    private final String TAIL_CHECKPOINT = "checkpoint.";
+
+    public FileCheckpointIO(String dirPath) {
+        this.dirPath = dirPath;
+    }
+
+    @Override
+    public Checkpoint read(String fileName) throws IOException {
+        Path path = Paths.get(dirPath, fileName);
+        InputStream is = Files.newInputStream(path);
+        return read(new BufferedChecksumStreamInput(new InputStreamStreamInput(is)));
+    }
+
+    @Override
+    public Checkpoint write(String fileName, int pageNum, int firstUnackedPageNum, long firstUnackedSeqNum, long minSeqNum, int elementCount) throws IOException {
+        Path path = Paths.get(dirPath, fileName);
+        Checkpoint checkpoint = new Checkpoint(pageNum, firstUnackedPageNum, firstUnackedSeqNum, minSeqNum, elementCount);
+        final byte[] buffer = new byte[BUFFER_SIZE];
+        write(checkpoint, buffer);
+        Files.write(path, buffer);
+        return checkpoint;
+    }
+
+    @Override
+    public void purge(String fileName) throws IOException {
+        Path path = Paths.get(dirPath, fileName);
+        Files.delete(path);
+    }
+
+    @Override
+    public void purge() throws IOException {
+        // TODO: dir traversal and delete all checkpoints?
```

Comment:
need to decide if this is a merge blocked, I don't think it is ... the purging fucntionality is not all fleshed out